pkg/runctl/run_manager.go

```diff
@@ -457,7 +461,14 @@ func (c *runManager) getServiceAccountSecretName(ctx *runContext) string {
 		return c.testing.getServiceAccountSecretNameStub(ctx)
 	}
 
-	return ctx.serviceAccount.GetHelper().GetServiceAccountSecretNameRepeat()
+	start := time.Now()
+	secretName := ctx.serviceAccount.GetHelper().GetServiceAccountSecretNameRepeat()
+	end := time.Now()
+	elapsed := end.Sub(start)
+	klog.V(6).Infof("getServiceAccountSecretName for %q took %v", ctx.pipelineRun.String(), elapsed)
+
+	c.metrics.ObserveRetryDurationByType("ServiceAccountSecret", elapsed)
+	return secretName
```

Comment:
The type name `ServiceAccountSecret` is not self-explanatory. You have to know the implementation to understand what is meant. Assuming we use this metric for _many_ code places with retries it becomes hard to give each of them a good name.

Instead of inventing names for the retry location, we could simply use the program structure itself. The observation function could obtain information about its caller and use it for labeling. One label could specify only the direct caller, while another label could include the top N frames of the call stack. See [`runtime.Callers()`](https://pkg.go.dev/runtime#Callers) and [`runtime.CallFrames()`](https://pkg.go.dev/runtime#CallersFrames) for details.

Reply:
I agree to get a better name but I think using Callers/CallFrames is not the way to go.

Reply:
Changed